app/models/user.php

```diff
@@ -76,4 +95,59 @@ public static function getUsername($user_id)
 
         return $row['username'];
     }
+
+    public static function getOwnUserDetails($user_id)
+    {
+        $db = DB::conn();
+
+        $row = $db->row('SELECT user_id, username, firstname, lastname, email, department FROM user WHERE user_id= ?',
+        array($user_id));
+
+        if (!$row) {
+            throw new RecordNotFoundException('No record found');
+        }
+            return new self($row);
+    }
+
+    public function updateUserDetails($user_id)
+    {
+
+        if (!$this->validate()) {
+            throw new ValidationException('Oops! invalid credentials');
+        }
+
+        $db = DB::conn();
+
+        $params = array(
+            'username'   => $this->username,
+            'firstname'  => $this->firstname,
+            'lastname'   => $this->lastname,
+            'email'      => $this->email,
+            'department' => $this->department
+        );
+
+        $where_params = array(
+            'user_id' => $user_id
+        );
+
+        try {
+            $db->update('user', $params, $where_params);
+        } catch (PDOException $e) {
+            if ($e->errorInfo[1] === self::MYSQL_ERROR_CODE) {
+                throw new DuplicateEntryException('Duplicate Entry');
+            }
+        }
+    }
+
+    public function updateUserPassword()
```

Comment:
for ```updateUserDetails```, please see [this](https://github.com/maxcarolino/Forum/pull/2/files#r40164798)